paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerSetSpawnScriptEvent.java

```diff
@@ -0,0 +1,108 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import com.destroystokyo.paper.event.player.PlayerSetSpawnEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import static com.denizenscript.denizen.paper.PaperModule.componentToJson;
+import static com.denizenscript.denizen.paper.PaperModule.jsonToComponent;
+
+public class PlayerSetSpawnScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player spawn changes
+    //
+    // @Cancellable true
+    //
+    // @Location true
```

Comment:
Meta has this, but `matches` doesn't seem to actually have a location check - I'd probably do it on the new spawn position?